net/net.go

```diff
@@ -16,7 +16,71 @@ limitations under the License.
 
 package net
 
-import "net"
+import (
+	"fmt"
+	"net"
+)
+
+// ParseCIDRsStrings parses a list of cidrs and return error if any is in valid.
+// order is maintained
+func ParseCIDRsStrings(cidrsString []string) ([]*net.IPNet, error) {
+	cidrs := make([]*net.IPNet, len(cidrsString))
+	for idx, current := range cidrsString {
+		_, cidr, err := net.ParseCIDR(current)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cidr value:%q with error:%v", current, err)
+		}
+		cidrs[idx] = cidr
+	}
+	return cidrs, nil
+}
+
+// IsDualStackIPs returns if ips are <v4>,<v6> or <v6>,<v4>
+func IsDualStackIPs(ips []net.IP) bool {
+	if len(ips) != 2 {
+		return false
+	}
+	if ips[0] == nil || ips[1] == nil {
+		return false
+	}
+	return IsIPv6(ips[0]) != IsIPv6(ips[1])
+}
+
+// IsDualStackIPsStrings returns if ips are (string) are <v4>,<v6> or <v6>,<v4>
+func IsDualStackIPsStrings(ips []string) bool {
+	if len(ips) != 2 {
+		return false
+	}
+	ip0 := net.ParseIP(ips[0])
+	ip1 := net.ParseIP(ips[1])
+
+	if ip0 == nil || ip1 == nil {
```

Comment:
isn't this redundant given what IsDualStackIPs does?

Reply:
it is a bit. But i'd rather have the check done instead of `IsDualStackIPs` changes and we forget to update ..